encryptor/queryDataEncryptor.go

```diff
@@ -0,0 +1,263 @@
+					}
+					return false, nil
+				}
+				if err := base.ValidateAcraStructLength(rawData); err == nil {
+					logrus.Debugln("Skip encryption for matched AcraStruct structure")
+					return false, nil
+				}
+				encrypted, err := encryptor.encryptWithColumnSettings(schema.GetColumnEncryptionSettings(columnName), rawData)
+				if err != nil {
+					logrus.WithError(err).Errorln("Can't encrypt hex value from query")
+					return false, err
+				}
+				encoded, err := encryptor.dataCoder.Encode(val, encrypted)
+				if err != nil {
+					if err != errUnsupportedExpression {
+						logrus.WithError(err).Errorln("Can't encode data")
+					}
+					return false, err
+				}
+				val.Val = encoded
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
+// tableData store TableName and related As value together
+type tableData struct {
+	TableName sqlparser.TableName
+	As        sqlparser.TableIdent
+}
+
+// getTablesFromUpdate collect all tables from all update TableExprs which may be as subquery/table/join/etc
+// collect only table names and ignore aliases for subqueries
+func (encryptor *QueryDataEncryptor) getTablesFromUpdate(tables sqlparser.TableExprs) []*tableData {
+	var outputTables []*tableData
+	for _, tableExpr := range tables {
+		switch statement := tableExpr.(type) {
+		case *sqlparser.AliasedTableExpr:
+			aliasedStatement := statement.Expr.(sqlparser.SimpleTableExpr)
+			switch simpleTableStatement := aliasedStatement.(type) {
+			case sqlparser.TableName:
+				outputTables = append(outputTables, &tableData{TableName: simpleTableStatement, As: statement.As})
+			case *sqlparser.Subquery:
+				// unsupported
+			default:
+				logrus.Debugf("Unsupported SimpleTableExpr type %s", reflect.TypeOf(simpleTableStatement))
+			}
+		case *sqlparser.ParenTableExpr:
+			outputTables = append(outputTables, encryptor.getTablesFromUpdate(statement.Exprs)...)
+		case *sqlparser.JoinTableExpr:
+			outputTables = append(outputTables, encryptor.getTablesFromUpdate(sqlparser.TableExprs{statement.LeftExpr, statement.RightExpr})...)
+		default:
+			logrus.Debugf("Unsupported TableExpr type %s", reflect.TypeOf(tableExpr))
+		}
+	}
+	return outputTables
+}
+
+// hasTablesToEncrypt check that exists schema for any table in tables
+func (encryptor *QueryDataEncryptor) hasTablesToEncrypt(tables []*tableData) bool {
+	for _, table := range tables {
+		if v := encryptor.schemaStore.GetTableSchema(table.TableName.Name.String()); v != nil {
+			return true
+		}
+	}
+	return false
+}
+
+// encryptUpdateExpressions try to encrypt all supported exprs. Use firstTable if column has not explicit table name because it's implicitly used in DBMSs
+func (encryptor *QueryDataEncryptor) encryptUpdateExpressions(exprs sqlparser.UpdateExprs, firstTable sqlparser.TableName, qualifierMap qualifierToTableMap) (bool, error) {
+	var schema *TableSchema
+	changed := false
+	for _, expr := range exprs {
+		// recognize table name of column
+		if expr.Name.Qualifier.IsEmpty() {
```

Comment:
first table is default table?

Reply:
yes. if multi-table update then if column without any table as prefix then work as it column from first table in the list